internal/policy/verify.go

```diff
@@ -129,6 +143,38 @@ func VerifyRelativeForRef(ctx context.Context, repo *git.Repository, initialPoli
 	return nil
 }
 
+// VerifyNewPolicyUsingExistingPolicy ensures that when a new policy is
+// encountered, its root role is signed by keys trusted in the prior policy.
+func VerifyNewPolicyUsingExistingPolicy(ctx context.Context, newPolicy, currentPolicy *State) error {
```

Comment:
Would this make sense to define as a receiver on State?

e.g.

```go
func (s *State) VerifyState(ctx context.Context, new *State) error {
  ...
}
```

Reply:
Done!